src/custom_connect.ts

```diff
@@ -2,7 +2,8 @@ import net from 'net';
 import type http from 'http';
 import { promisify } from 'util';
 
-const asyncWrite = promisify(net.Socket.prototype.write);
+const socket = new net.Socket();
+const asyncWrite = promisify(socket.write).bind(socket);
```

Comment:
This is unnecessarily hacky, let's call `promisify` in `customConnect` function, no need to do it here (performance impact is minuscule)

Reply:
Right, I didn't realize that `customConnect` receives a Socket instance. Updated!